### PR TITLE
PD-286: Logo Component

### PR DESCRIPTION
**Pull request by brookescarlett**
## ✍️ Proposed changes

- Created Logo component, which exports { Logo, LogoMark } 
- Logo available as knockout: true/false, and variant: dark/light
- LogoMark available as knockout: true/false, variant: dark/light. If no variant is set, LogoMark will be rendered full-color

Note: the component is fully ready for review from an API and implementation perspective, but I want to replace the existing SVGs here with ones with less width built-in. As-is there's a lot of extra width, which will throw off consumers when they try and implement locally. (This is visible in the screenshots, where I've applied a padding of 10px in every direction to the container div and the width of the div far exceeds the width of the Logo/LogoMark components)

🎟 _Jira ticket:_ [PD-286](https://jira.mongodb.org/browse/PD-286)

## 🛠 Types of changes

- [ ] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [x] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## ✅ Checklist

- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have added necessary documentation (if appropriate)
- [x] I have run `yarn changeset` and documented my changes

Screen Shot 2019-11-14 at 10 01 13 AM
Screen Shot 2019-11-14 at 10 01 16 AM
Screen Shot 2019-11-14 at 10 01 23 AM

### CODE REVIEW

File: packages/logo/README.md [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+
+<Logo />
+```
+
+**Output HTML**
+
+```HTML
+<svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 1112.61 300" height="40" width="100%">
+  <title>MongoDB Logo</title>
+  <path class="leafygreen-ui-17842x1" d="M134.44,120.34C119.13,52.8,87.22,34.82,79.08,22.11a144.57,144.57,0,0,1-8.9-17.42c-.43,6-1.22,9.78-6.32,14.33C53.62,28.15,10.13,63.59,6.47,140.33c-3.41,71.55,52.6,115.67,60,120.23,5.69,2.8,12.62.06,16-2.51,27-18.53,63.89-67.93,52-137.71"></path>
```
Comment (DesignerDave) [inline packages/logo/README.md]: Let's edit out all the path attribute values.

```
d="..."
```

File: packages/logo/README.md [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+
+<LogoMark />
+```
+
+**Output HTML**
+
+```HTML
+<svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 216.56 448.5" height="40" width="auto">
+  <title>MongoDB LogoMark</title>
+  <path class="leafygreen-ui-17842x1" d="M202.8,179.68c-23-101.47-71-128.49-83.18-147.59C113,21.7,106.25,5.91,106.25,5.91c-.66,9-1.83,14.7-9.51,21.54C81.36,41.16,16,94.42,10.51,209.72c-5.12,107.5,79,173.8,90.18,180.65,8.54,4.2,19,.08,24-3.77,40.54-27.84,96-102.07,78.06-206.92"></path>
```
Comment (DesignerDave) [inline packages/logo/README.md]: Ditto here to my path value comment.

File: packages/logo/src/Logo.tsx [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+ */
+function Logo({ variant = 'dark', knockout = false, height = 40 }: LogoProps) {
+  let fill = '';
+
+  const className = css`
+    width: auto;
+    height: ${height}px;
+  `;
+
+  if (variant === 'light') {
```
Comment (DesignerDave) [inline packages/logo/src/Logo.tsx]: This conditional block feels redundant. Why not:
```js
const fill = variant === 'light' ? uiColors.white : uiColors.gray.dark3;

if (knockout) {
  return <MonochromeLogo className={className} fill={fill} />;
}

return <RGBLogo className={className} fill={fill} />;
```

Comment (DesignerDave) [reply packages/logo/src/Logo.tsx]: Better yet, we can just mirror how LogoMark renders the component.

File: packages/logo/src/logos/MonochromeLogoMark.tsx [hunk 4]
```diff
@@ -0,0 +1,19 @@
+import React from 'react';
+import { SVGLogoProps } from '../types';
+
+export default function MonochromeLogoMark({ fill, className }: SVGLogoProps) {
+  return (
+    <svg
+      xmlns="http://www.w3.org/2000/svg"
+      viewBox="0 0 216.56 448.5"
+      data-testid={`${fill}-monochrome-logomark`}
```
Comment (DesignerDave) [inline packages/logo/src/logos/MonochromeLogoMark.tsx]: We shouldn't render this in our component. If we spread `rest`, we can let the test pass this ID.

File: packages/logo/src/logos/MonochromeLogoMark.tsx [hunk 5]
```diff
@@ -0,0 +1,19 @@
+import React from 'react';
+import { SVGLogoProps } from '../types';
+
+export default function MonochromeLogoMark({ fill, className }: SVGLogoProps) {
+  return (
+    <svg
+      xmlns="http://www.w3.org/2000/svg"
```
Comment (DesignerDave) [inline packages/logo/src/logos/MonochromeLogoMark.tsx]: I don't think SVGs need this anymore, but could be wrong.

File: packages/logo/src/logos/RGBLogo.tsx [hunk 6]
```diff
@@ -0,0 +1,72 @@
...
+
+const class1 = css`
+  fill: #10aa50;
+`;
+
+const class2 = css`
+  fill: #b8c4c2;
+`;
+
+const class3 = css`
```
Comment (DesignerDave) [inline packages/logo/src/logos/RGBLogo.tsx]: These feel very appropriate to inline.

File: packages/logo/README.md (hunk 1 above)
Comment (brookescarlett) [reply packages/logo/README.md]: 👍

File: packages/logo/README.md (hunk 2 above)
Comment (brookescarlett) [reply packages/logo/README.md]: 👍

File: packages/logo/src/Logo.tsx (hunk 3 above)
Comment (brookescarlett) [reply packages/logo/src/Logo.tsx]: 👍

File: packages/logo/src/logos/MonochromeLogoMark.tsx (hunk 4 above)
Comment (brookescarlett) [reply packages/logo/src/logos/MonochromeLogoMark.tsx]: 👍

File: packages/logo/src/logos/MonochromeLogoMark.tsx (hunk 5 above)
Comment (brookescarlett) [reply packages/logo/src/logos/MonochromeLogoMark.tsx]: 👍

File: packages/logo/src/Logo.story.tsx [hunk 7]
```diff
@@ -0,0 +1,59 @@
...
+
+storiesOf('Logo', module)
+  .add('LogoMark', () => {
+    const variant = select(
+      'Variant',
+      Object.values(Variant) as Array<Variant>,
+      Variant.Dark,
+    );
+
+    console.log(variant);
```
Comment (hswolff) [inline packages/logo/src/Logo.story.tsx]: rm

File: packages/logo/src/Logo.spec.tsx [hunk 8]
```diff
@@ -0,0 +1,77 @@
...
+    test('renders dark knockout logomark, when variant is set to dark', () => {
+      const testId = '#21313C-monochrome-logomark';
+      const { getByTestId } = render(
+        <LogoMark variant="dark" knockout={true} data-testid={testId} />,
+      );
+      const logomark = getByTestId(testId);
+
+      expect(logomark).toBeInTheDocument();
+    });
+    test('renders light knockout logomark, when variant is set to light', () => {
```
Comment (hswolff) [inline packages/logo/src/Logo.spec.tsx]: keep breaklines between test blocks consistent please!

Comment (brookescarlett) [reply packages/logo/src/Logo.spec.tsx]: 👍

File: packages/logo/src/Logo.story.tsx (hunk 7 above)
Comment (brookescarlett) [reply packages/logo/src/Logo.story.tsx]: 👍

**Pull request merged**